CMake/macOSMaxAllowed.c

```diff
@@ -0,0 +1,8 @@
+#include <AvailabilityMacros.h>
+#include <stdio.h>
+
+int main(int argc, char **argv) {
+  unsigned int ver = __MAC_OS_X_VERSION_MAX_ALLOWED;
+  printf("%d.%d.%d", ver/10000, (ver/100)%100, ver%100);
```

Comment:
Same as above, should it fail if the "underscored" macro is not defined?

Reply:
Fixed the include file, failure calls a reasonable fallback. Tested.